src/Catel.Core/Catel.Core.Shared/Data/ModelBase.cs

```diff
@@ -312,6 +313,19 @@ protected internal bool LeanAndMeanModel
         public static bool DefaultDisableEventSubscriptionsOfChildValuesValue { get; set; }
 
         /// <summary>
+        /// Gets or sets a value indicating whether usage of equality comparer for equals should be disabled.
+        /// <para />
+        /// The default value is <c>false</c>.
+        /// </summary>
+        /// <value><c>true</c> if usage of equality comparer for equals should be disabled; otherwise, <c>false</c>.</value>
+        protected bool DisableUsageOfEqualityComparerForEquals { get; set; }
```

Comment:
GetType() gets worse if it gets called too many times, so yes. Befor I implement a caching mechanism, let us discuss an alternative for your or someone others needs of having this ModelEqualityComparer in Equals():
Remove the whole thing and if somebody wants it he should use the comparer directly or pass it into collections which have for exact this reason an extra comparer parameter! And if someone wants to have this by default he can override Equals(), as I did to remove it, and implement this by its own. I believe this is more the .NET way. You know what I mean? I am really not happy that you have altered the basics of .NET and we had performance issues for over two years because we did not get as deep in Catel as we are now and without a profiler.